client/components/stats-date-control/stats-date-control-date-input.tsx

```diff
@@ -18,7 +21,42 @@ const DateInput: React.FC< Props > = ( {
 		onChange && onChange( event.target.value );
 	};
 
-	return <input id={ id } type="date" value={ value } onChange={ handleChange } max={ max } />;
+	const toDateString = ( date: moment.MomentInput ) => {
+		if ( moment.isMoment( date ) || moment( date ).isValid() ) {
+			return moment( date ).format( 'YYYY-MM-DD' );
+		}
+		return moment().format( 'YYYY-MM-DD' );
+	};
+
+	const handleBlur = () => {
+		if ( isNewCalendar ) {
+			const isValid = moment( value, 'YYYY-MM-DD', true ).isValid();
+			if ( ! isValid ) {
+				onChange && onChange( moment().format( 'YYYY-MM-DD' ) );
+			} else {
+				toDateString( value );
+			}
+		}
+	};
+
+	return (
+		<>
+			{ isNewCalendar ? (
+				<input
+					id={ id }
+					type="text"
+					value={ value }
+					onChange={ handleChange }
+					onBlur={ handleBlur } // Add onBlur event for validation
```

Comment:
Since we have both onChange and onBlur, input value will be set for the whole component, also the calendar that uses the property. When users type a letter, the page breaks.

Either we have to:
- use an internal state and trigger an internal change event to track the input value and trigger the external onChange (from the props) when the blur event successfully sets a valid date or 
- we have to keep updating the value that is being typed as long as it's a number and reject letters, making sure that the calendar stops updating in the calendar file